inc/sainexthopgroup.h

```diff
@@ -184,6 +184,41 @@ typedef enum _sai_next_hop_group_attr_t
      */
     SAI_NEXT_HOP_GROUP_ATTR_SELECTION_MAP,
 
+    /**
+     * @brief Adaptive routing and switching object for this group.
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_ARS
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_NEXT_HOP_GROUP_ATTR_ARS_OBJECT_ID,
+
```

Comment:
Can we consider the use of different ARS objects (profiles) for the same ECMP ? For example, different type of traffic (UDP vs TCP) can be most efficiently used with different ARS modes - e.g. flowlet for TCP to minimize re-ordering and packet_random for UDP when no fear of reordering. For that we can use the ACL action with indication of this ARS object to use or using some enumeration of ARS/profiles. Then the TCP/UDP traffic can be identified with ACL match and the corresponding ARS object to be used

Reply:
Granular control for TCP or UDP traffic is managed by creating an ACL rule with action as  ARS object id.
Nexthopgroup object will have reference to only single ARS object id.